src/processors/processPostRequest.ts

```diff
@@ -0,0 +1,114 @@
+import { isValidObject } from '@dvsa/cvs-type-definitions/schema-validator';
+import logger from '../util/logger';
+import { generateNewNumber, NumberTypes } from '../services/testNumber';
+import { identifySchema } from '../validators/post';
+import { UserDetails } from '../services/user';
+import { HttpMethod, RecordCompleteness, VehicleType } from '../util/enum';
+import { TechrecordGet, TechrecordPut } from '../models/post';
+
+export const processPostRequest = async (input: unknown, userDetails: UserDetails) => {
+  // we should be validating it's a valid technical record HERE.
```

Comment:
added a comment. We should be validating it's a valid technical record here. 
then we can use typescript's [type predicate system](https://www.typescriptlang.org/docs/handbook/advanced-types.html#using-type-predicates) such as

```ts
const isTechRecord(input): input is TechrecordPut => {
  return isThisObjectValid(object)
}
```

Reply:
I assume a valid technical record would just mean that it matches the skeleton schema of whichever vehicle we're trying to send.

Reply:
I'm happy to add that in but based on a previous comment which was on the computeRecordCompleteness method, you mentioned that if it's not a complete or testable record then we can assume that it is a skeleton record. Is it a case of that, that I need to move the computeRecordCompleteness value to be the first? Or do I need to add back the logic to test that the tech record is a skeleton one and if it isn't throw some kind of invalid tech record error?